lib/adapters/http.js

```diff
@@ -250,16 +254,22 @@ module.exports = function httpAdapter(config) {
     });
 
     // Handle request timeout
+    //
+    // There are many types of timeouts:
+    // DNS Timeout, Connect Timeout, Read/Write Timeout (idle timeout), and Call Timeout.
+    // Call Timeout measures from DNS Lookup to response body receiving completion.
+    //
+    // Currently, `timeout` option in Browser XHR indicates "Call Timeout".
+    // Create internal timer to match browser-side behavior.
     if (config.timeout) {
-      // Sometime, the response will be very slow, and does not respond, the connect event will be block by event loop system.
-      // And timer callback will be fired, and abort() will be invoked before connection, then get "socket hang up" and code ECONNRESET.
-      // At this time, if we have a large number of request, nodejs will hang up some socket on background. and the number will up and up.
-      // And then these socket which be hang up will devoring CPU little by little.
-      // ClientRequest.setTimeout will be fired on the specify milliseconds, and can make sure that abort() will be fired after connect.
-      req.setTimeout(config.timeout, function handleRequestTimeout() {
```

Comment:
If dropping connect timeout, users will complain about server problems that fixed by #1752. Maybe we should consider #1739 together, and [got#timeout](https://github.com/sindresorhus/got#timeout) already gives a good example.

To keep things simple, we can only support partial kinds of timeout at the beginning.

Reply:
Actually, `HTTPRequest#setTimeout` cannot detect connect timeout. I'm not sure it's intended or just a bug of Node.js core. Current `HTTPRequest#setTimeout` implementation won't call `Socket#setTimeout` internally if connect timeout was happened. Please refer to https://github.com/nodejs/node/blob/v13.12.0/lib/_http_client.js#L808-L812 . You'll notice `Socket#setTimeout` will not be called until connection establish.

Reply:
Here's a proof that `HTTPRequest#setTimeout` cannot detect connect timeout due to described reason:

<img width="851" alt="스크린샷 2020-04-06 오전 5 57 29" src="https://user-images.githubusercontent.com/2101743/78509813-c7a88d00-77cb-11ea-88ef-11a689cefadf.png">
<img width="743" alt="스크린샷 2020-04-06 오전 5 58 59" src="https://user-images.githubusercontent.com/2101743/78509814-c8d9ba00-77cb-11ea-9a64-39ac16138dd7.png">

It results to axios too. latest axios release (which is #1752 changes applied) cannot detect connect timeout either:

<img width="712" alt="스크린샷 2020-04-06 오전 5 52 22" src="https://user-images.githubusercontent.com/2101743/78509807-bfe8e880-77cb-11ea-8229-a77cac37528a.png">




  axios cannot detect connect timeout due to described reason.

As you can see. axios v0.19.2 cannot detect connect timeout and elapsed 1 minutes (macOS system default connect timeout)

<img width="712" alt="스크린샷 2020-04-06 오전 5 52 22" src="https://user-images.githubusercontent.com/2101743/78509690-eeb28f00-77ca-11ea-8295-893f97356971.png">